
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>Fundamental Concepts &#8212; DELINEATE 1.2.3 documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/language_data.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Quick Start" href="quickstart.html" />
    <link rel="prev" title="Installation" href="install.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <div class="section" id="fundamental-concepts">
<h1>Fundamental Concepts<a class="headerlink" href="#fundamental-concepts" title="Permalink to this headline">¶</a></h1>
<div class="section" id="partitions">
<h2>“Partitions”<a class="headerlink" href="#partitions" title="Permalink to this headline">¶</a></h2>
<p>The primary output of DELINEATE is essentially calculation of the probabilities of different <em>partitions</em> of the population lineages in the study into distinct species.
Each partition represents a different collection of assignments of species identities to each of the population lineages.
The term “partition” here comes from basic set theory: it is a division of a set of elements into a set of <em>mutually exclusive</em> and <em>jointly comprehensive</em> subsets.
In this application, we are considering the population lineages (i.e., the tips of the population tree we give as input to DELINEATE) as the elements of the initial set.
We want to organize this set of populations into distinct subsets, where each subset corresponds to a distinct species.</p>
<p>For example, imagine that that we have a system consisting of three populations, A, B, and C, which can be represented by the following set of three elements:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span><span class="n">A</span><span class="p">,</span> <span class="n">B</span><span class="p">,</span> <span class="n">C</span><span class="p">}</span>
</pre></div>
</div>
<p>What are the different “species delimitations” are possible with this system?
That is, what are the different ways we can we organize this set of populations in to species?</p>
<blockquote>
<div><ol class="arabic">
<li><p>One way would be to lump all the populations to the same species, i.e.:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{{</span><span class="n">A</span><span class="p">,</span><span class="n">B</span><span class="p">,</span><span class="n">C</span><span class="p">}}</span>
</pre></div>
</div>
</li>
<li><p>Another way might be to split off each population to its own distinct species, i.e.:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{{</span><span class="n">A</span><span class="p">},{</span><span class="n">B</span><span class="p">},{</span><span class="n">C</span><span class="p">}}</span>
</pre></div>
</div>
</li>
<li><p>Another way might be to have A and B assigned to the same species, but C to a different one:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{{</span><span class="n">A</span><span class="p">,</span><span class="n">B</span><span class="p">},{</span><span class="n">C</span><span class="p">}}</span>
</pre></div>
</div>
</li>
<li><p>Or, alternatively, A and C to the same species, but B to a different one:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{{</span><span class="n">A</span><span class="p">,</span><span class="n">C</span><span class="p">},{</span><span class="n">B</span><span class="p">}}</span>
</pre></div>
</div>
</li>
<li><p>Or, finally, B and C to the same species, but A to a different one:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{{</span><span class="n">A</span><span class="p">},{</span><span class="n">B</span><span class="p">,</span><span class="n">C</span><span class="p">}}</span>
</pre></div>
</div>
</li>
</ol>
</div></blockquote>
<p>Each of these five arrangements is a distinct partition of the original set of three elements.
Thus, each possible partition of set of population lineages represents a different way of organizing that set of population lineages into species.
DELINEATE will report the probabilities of each of these partitions for your data.
The partition with the highest probability represents the maximum likelihood estimate (MLE) of the species identities of the populations in your system.</p>
</div>
<div class="section" id="the-diversification-process">
<h2>The Diversification Process<a class="headerlink" href="#the-diversification-process" title="Permalink to this headline">¶</a></h2>
<p>How are the probabilities of these different partitions of population lineages calculated?
To understand that we have to understand how DELINEATE models the diversification process.
DELINEATE models diversification with the following events:</p>
<ul class="simple">
<li><p><em>population splitting</em></p></li>
<li><p><em>population extinction</em></p></li>
<li><p><em>evolution of reproductive isolation of an independent population lineage</em></p></li>
</ul>
<p>Population splitting, i.e. the fragmentation of an ancestral population lineage into two independent daughter lineages, corresponds to a birth event on a tree of populations (from one or more species) .
Each of the two daughter lineages proceeds on its own independent trajectory, still part of the same species as the parent population lineages, until they either themselves split into further daughter lineages, go extinct (corresponding to death events on a tree of populations), or develop into a distinct species (from either parent or sister population).
As modeled by DELINEATE the formation of new independent population lineages (birth events on a tree), the loss of these lineages due to extinction (death events on a tree), and the development of a lineage into a full independent species, are modeled as three independent processes, each with their own distinct rate.
Birth events on a tree can be considered the (potential) origin of a new species, and the the development of full independent species status as the completion of this phenomenon.
As such, with DELINEATE speciation is not an instantaneous event (such as that modeled by a simple birth-death branching model underlying a species tree), but an extended process, starting with the lineeage splitting and ending with the speciation completion event.</p>
<p>We can consider the speciation completion process to “play out” over a birth-death tree, much in the same way as character evolution might occur on a phylogeny.</p>
<div class="figure align-default">
<img alt="_images/diversification.png" src="_images/diversification.png" />
</div>
</div>
<div class="section" id="sampling-design">
<h2>Sampling Design<a class="headerlink" href="#sampling-design" title="Permalink to this headline">¶</a></h2>
<p>DELINEATE requires a fundamentally different way to thinking how we sample data for species delimitation studies.</p>
<p>DELINEATE ideally should be provided with data that <em>includes samples from as many populations as possible across the system</em> being studied. This is in contrast from standard practice from other approaches, in which typically one or two examplar population sample per putative species are included. The theoretical ideal would be to include <em>every</em> population of all species in the system, known or unknown, i.e., to capture all population isolation or splitting events. Of course, we do not expect to achieve this theoretical ideal in practice, but it is certainly something to aspire to. The key point is restricting our population/species sampling to a few examplar populations per species is something we want to move away from.</p>
<p>DELINEATE also requires that we have know the species identities of at least <em>some</em> of our population lineages. This is, again, in contrast to other approaches to species delimitation, which might be quite happy analyzing an entire data set with no known fixed species identities. With DELINEATE we should design our sampling scheme to including a much broader range of species than just the ones we are interested in delimiting, and should include populations belong to species in which we are quite confident regarding their species identities. These other species — or, to be more precise, population lineages for which the species identities are known — are critical to allowing DELINEATE to “learn” about the speciation process.</p>
</div>
<div class="section" id="from-individuals-to-populations-to-species">
<h2>From Individuals to Populations to Species<a class="headerlink" href="#from-individuals-to-populations-to-species" title="Permalink to this headline">¶</a></h2>
<p>The DELINEATE package itself represents the final step in an analytical pipeline.
Starting with a collection of genetic alignments representing multiple genes sampled from multiple individuals from multiple populations, each step of the pipeline groups the data into successively higher levels of organization, from populations to species.</p>
<p>A typical species delimitation analysis would consist of the following three steps:</p>
<blockquote>
<div><ol class="arabic simple">
<li><p><strong>Identification (Delimitation) of *Population* Units:</strong> First, we would carry out a <a class="reference external" href="https://github.com/bpp/bpp">BP&amp;P</a> analysis to identify population units by aggregating individuals into populations under the multipopulation coalescent model. We would typically hope to sample at least a few genes from two to ten individuals per population, with multiple populations per putative species.
We would then use <a class="reference external" href="https://github.com/bpp/bpp">BP&amp;P</a> to organize these individuals into populations.
Note that <a class="reference external" href="https://github.com/bpp/bpp">BP&amp;P</a> terminology uses the term “species” and “populations” interchangeably. This can be confusing, but it is important to keep this in mind.</p></li>
<li><p><strong>Organization of the Population Units into a Population Phylogeny:</strong> Then, we would carry out a <a class="reference external" href="https://taming-the-beast.org/tutorials/starbeast2-tutorial/">StarBeast2</a> analysis using the groupings identified by BP&amp;P as “species”, to estimate an ultrametric phylogeny with those groupings as tips (i.e, a population phylogeny).
Once we have decided what our population units are, we will use <a class="reference external" href="https://taming-the-beast.org/tutorials/starbeast2-tutorial/">StarBeast2</a> to infer an ultrametric population tree to use as input. Here, again, while StarBeast2 uses the terminology “species” to reference to groupings of individuals, we should bear in mind that we are still dealing with population. We will use the units identified as populations by BPP as the “species” grouping in StarBeast2.</p></li>
<li><p><strong>Calculating the Probability of Species Assignments:</strong> Finally, the actual species delimitation analysis itself: a DELINEATE analysis to calculate the probabilities of different groupings of population tips of the population tree into species.
The population tree resulting from <a class="reference external" href="https://taming-the-beast.org/tutorials/starbeast2-tutorial/">StarBeast2</a> forms the one of the mandatory inputs for DELINEATE. The species identities for the subset of population lineages for which these are known forms the other. Running DELINEATE will then report on the probabilities of different species assignments for the remaining lineages, i.e. for the ones for which we do not know or specifies the species identities.</p></li>
</ol>
</div></blockquote>
</div>
</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">DELINEATE</a></h1>








<h3>Navigation</h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="install.html">Installation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Fundamental Concepts</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#partitions">“Partitions”</a></li>
<li class="toctree-l2"><a class="reference internal" href="#the-diversification-process">The Diversification Process</a></li>
<li class="toctree-l2"><a class="reference internal" href="#sampling-design">Sampling Design</a></li>
<li class="toctree-l2"><a class="reference internal" href="#from-individuals-to-populations-to-species">From Individuals to Populations to Species</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="quickstart.html">Quick Start</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="install.html" title="previous chapter">Installation</a></li>
      <li>Next: <a href="quickstart.html" title="next chapter">Quick Start</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2020, Jeet Sukumaran and Mark T. Holder.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 2.4.0</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/concepts.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>