

<!doctype html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>A Complete Worked Example: Lionepha &#8212; DELINEATE 1.2.3 documentation</title>
    <link rel="stylesheet" href="_static/bizstyle.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/css/docs.css" />
    
    <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/language_data.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script src="_static/bizstyle.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Quick Start" href="quickstart.html" />
    <meta name="viewport" content="width=device-width,initial-scale=1.0">
    <!--[if lt IE 9]>
    <script src="_static/css3-mediaqueries.js"></script>
    <![endif]-->
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="quickstart.html" title="Quick Start"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">DELINEATE 1.2.3 documentation</a> &#187;</li> 
      </ul>
    </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h3><a href="index.html">Table of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="install.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="concepts.html">Basic Concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="quickstart.html">Quick Start</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">A Complete Worked Example: <em>Lionepha</em></a><ul>
<li class="toctree-l2"><a class="reference internal" href="#software-prerequisites">(Software Prerequisites)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#data-files">(Data Files)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#introduction">Introduction</a></li>
<li class="toctree-l2"><a class="reference internal" href="#background">Background</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#study-design">Study Design</a></li>
<li class="toctree-l3"><a class="reference internal" href="#data">Data</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#stage-i-identification-of-population-units-using-the-multipopulation-coalescent-model">Stage I: Identification of Population Units using the Multipopulation Coalescent Model</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#candidate-population-units">Candidate Population Units</a></li>
<li class="toctree-l3"><a class="reference internal" href="#generating-a-guide-tree-for-population-delimitation">Generating a Guide Tree for Population Delimitation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#delimitation-of-population-units">Delimitation of Population Units</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#stage-ii-generating-the-multipopulation-coalescent-ultrametric-phylogeny-of-populations">Stage II. Generating the (Multipopulation Coalescent, Ultrametric) Phylogeny of Populations</a></li>
<li class="toctree-l2"><a class="reference internal" href="#stage-iii-assignment-of-known-vs-unknown-species-identities">Stage III. Assignment of Known vs. Unknown Species Identities</a></li>
<li class="toctree-l2"><a class="reference internal" href="#stage-iv-delimitation-of-species-units">Stage IV. Delimitation of Species Units</a></li>
<li class="toctree-l2"><a class="reference internal" href="#examination-and-interpretation-of-the-results">Examination and Interpretation of the Results</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="quickstart.html"
                        title="previous chapter">Quick Start</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/workflow1.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="a-complete-worked-example-lionepha">
<h1>A Complete Worked Example: <em>Lionepha</em><a class="headerlink" href="#a-complete-worked-example-lionepha" title="Permalink to this headline">¶</a></h1>
<div class="section" id="software-prerequisites">
<h2>(Software Prerequisites)<a class="headerlink" href="#software-prerequisites" title="Permalink to this headline">¶</a></h2>
<p>For this exercise, in addition to DELINEATE you will need the following programs installed on your execution host:</p>
<blockquote>
<div><ul class="simple">
<li><p><a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a></p></li>
<li><p><a class="reference external" href="https://taming-the-beast.org/tutorials/starbeast2-tutorial/"><span class="program">StarBeast2</span></a></p></li>
<li><p><a class="reference external" href="https://dendropy.org/"><span class="program">DendroPy</span></a> (which provides <a class="reference external" href="https://dendropy.org/programs/sumtrees.html"><span class="program">SumTrees</span></a>)</p></li>
</ul>
</div></blockquote>
<p>For meta-analysis and visualization, you should have the following installed on your local machine:</p>
<blockquote>
<div><ul class="simple">
<li><p><a class="reference external" href="http://tree.bio.ed.ac.uk/software/tracer/"><span class="program">Tracer</span></a></p></li>
<li><p><a class="reference external" href="http://tree.bio.ed.ac.uk/software/figtree/"><span class="program">FigTree</span></a></p></li>
</ul>
</div></blockquote>
<p>in addition, of course (and always), a <a class="reference external" href="https://sukumaranlab.org/resources/computational-skills/#a-full-featured-developer-s-text-editor-vim-emacs-or-sublime-text">*good* text editor</a>.</p>
</div>
<div class="section" id="data-files">
<h2>(Data Files)<a class="headerlink" href="#data-files" title="Permalink to this headline">¶</a></h2>
<p>All data files for this exercise are available for download here:</p>
<p class="downloadbox"><a class="reference download internal" download="" href="_downloads/c8d4870a23625b69cbc98b2e2143af01/lionepha.zip"><code class="xref download docutils literal notranslate"><span class="pre">lionepha.zip</span></code></a></p>
</div>
<div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>We are going to to carry out a full species delimitation analysis, or, actually, a series of species delimitation analyses.
We are going to use the the <em>Lionepha</em> system, using a subset of the data published by David R. Maddison and John S. Sproul:</p>
<ul class="simple">
<li><p>David R Maddison, John S Sproul. <em>Species delimitation, classical taxonomy and genome skimming: a review of the ground beetle genus Lionepha (Coleoptera: Carabidae)</em>, Zoological Journal of the Linnean Society, , zlz167, <a class="reference external" href="https://doi.org/10.1093/zoolinnean/zlz167">https://doi.org/10.1093/zoolinnean/zlz167</a>.</p></li>
</ul>
<p>The original data is available in the following <a class="reference external" href="https://datadryad.org">Dryad</a> repository:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://datadryad.org/stash/dataset/doi:10.5061/dryad.2jm63xsjq">https://datadryad.org/stash/dataset/doi:10.5061/dryad.2jm63xsjq</a></p></li>
</ul>
<p>The work by Maddison and Sproul (2020) nicely lays out the basic alpha taxonomy of this system, organizing these 143 lineages into 12 nominal species.
This alpha taxonomy is based on a thorough analyses integrating morphological as well as genomic information, and thus we already have a very clear understanding of the species boundaries in this group, precluding the need for a species delimitation analysis.</p>
<p>However, for the purposes of this exercise we are going to pretend that this is not the case.
Instead, we are going to consider that while the species identities of <em>some</em> of the population lineages are well known, in this sample there are number of lineages for which this is not the case.</p>
</div>
<div class="section" id="background">
<h2>Background<a class="headerlink" href="#background" title="Permalink to this headline">¶</a></h2>
<div class="section" id="study-design">
<h3>Study Design<a class="headerlink" href="#study-design" title="Permalink to this headline">¶</a></h3>
<p>We can imagine our study sampled individuals from an extensive range of localities, supplementing it with samples from previous collections, to develop a comprehensive data set spanning a broad taxonomic as well as geographic range.
Preliminary phylogenetic analysis, which included genomic data from known, stable, species, as well as comparative morphological analyses, has confidentally established the species identities of some of these lineages.
However, a number of lineages in the sample, while coming out to sister to known species, are sufficiently different in some respects (e.g., genetic distance or morphology), to make us question whether or not we are looking at different populations of the same species or perhaps new species altogether.</p>
<p>Alternatively, we might imagine a study to delineate species units in a particularly problematic clade.
Here, in addition to samples from multiple populations from the clade, we would also include populations from a broader range of species in which the focal clade is a nested subtree.
The analytical workflow for this sort of study will be identical to that discussed here, with only the distribution of population lineages of unknown identity being different — clustered into a single subtree on the larger phylogeny as opposed to scattered into multiple smaller subtrees (or single lineages).</p>
</div>
<div class="section" id="data">
<h3>Data<a class="headerlink" href="#data" title="Permalink to this headline">¶</a></h3>
<p>Here we will focus on just a subset of the data from Maddison and Sproul (2020), focusing only on lineages in the genus <em>Lionepha</em> itself.
This data consists of 8 gene alignments for 143 samples from multiple individuals from multiple populations from multiple species:</p>
<ul class="simple">
<li><p><span class="filepath">lionepha/00-alignments/18s.nex</span></p></li>
<li><p><span class="filepath">lionepha/00-alignments/28s.nex</span></p></li>
<li><p><span class="filepath">lionepha/00-alignments/argk.nex</span></p></li>
<li><p><span class="filepath">lionepha/00-alignments/cad.nex</span></p></li>
<li><p><span class="filepath">lionepha/00-alignments/coi.nex</span></p></li>
<li><p><span class="filepath">lionepha/00-alignments/msp.nex</span></p></li>
<li><p><span class="filepath">lionepha/00-alignments/topo.nex</span></p></li>
<li><p><span class="filepath">lionepha/00-alignments/wg.nex</span></p></li>
</ul>
</div>
</div>
<div class="section" id="stage-i-identification-of-population-units-using-the-multipopulation-coalescent-model">
<h2>Stage I: Identification of Population Units using the Multipopulation Coalescent Model<a class="headerlink" href="#stage-i-identification-of-population-units-using-the-multipopulation-coalescent-model" title="Permalink to this headline">¶</a></h2>
<p>The first stage of our analysis will be delimitation of <em>population</em> units, i.e. assignment of population identities to individual lineages, under the “multipopulation coalescent”.
This model was first published under the name “censored coalescent”, but has since become more popularly known as the “Multispecies Coalescent” or “MSC” model.
It is a very powerful and elegant model that allows us to identify structure in genomic data arising from disruptions in Wright-Fisher panmixia due to restrictions in gene flow.</p>
<p>We will be using <a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a> to identify our population units, and <a class="reference external" href="https://taming-the-beast.org/tutorials/starbeast2-tutorial/"><span class="program">StarBeast2</span></a> to obtain a guide tree for the <a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a> analysis.
Note that these programs use the term “species” to reference the higher-level units they analyze rather than populations (as in “species delimitation” or “species tree inference”, rather than “population delimitation” and “population tree inference”).
This can be correct in some contexts: e.g., where the data sampling does, either coincidentally or by design, indeed represent a 1-to-1 correspondence between populations and species, or, equivalently the species being studied do indeed <em>not</em> have any within-species lineage structure.
However, more generally, this is, at best, misleading.
These programs use the MSC as their underlying model.
From a statistical perspective, it is uncontroversial that the biological units that these programs model through the “censored” coalescent/MSC that model that underlies them are (Wright-Fisher) populations.
Strictly speaking, when used for species delimitation or model species trees, programs such as <a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a> and <a class="reference external" href="https://taming-the-beast.org/tutorials/starbeast2-tutorial/"><span class="program">StarBeast2</span></a> are used to identify <em>population</em> units under the MSC, and then these population units are interpreted (with or without justification) by the investigator to correspond on a 1-to-1 basis to species units.
Thus, when using the MSC for “species delimitation” or “species tree inference”, the difference between “population” and “species” is purely lexical rather than statistical.</p>
<p>Here, however, we are going to interpret higher-level units of organization as exactly what they are as modeled by the MSC — populations, no more and no less.
Unfortunately, this may result in some confusion as both <a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a> and <a class="reference external" href="https://taming-the-beast.org/tutorials/starbeast2-tutorial/"><span class="program">StarBeast2</span></a> refer to the higher-level units they target as “species” at least some of the time in the various program options and documentation.
(In fact, <a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a>, in some of the program documentation as well as n some of the various papers presenting or referencing the theory behind it acutally use the term “species” and “populations” interchangeably).
This is simply the cost of doing business.</p>
<div class="section" id="candidate-population-units">
<h3>Candidate Population Units<a class="headerlink" href="#candidate-population-units" title="Permalink to this headline">¶</a></h3>
<p>A <a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a> analysis requires us to identify “population” lineages as input <em>a priori</em>, some of which it will then collapse together to form “species” lineages.
As we have noted (and as we will note again!), this terminological choice is not generally correct (it <em>may</em> apply in some special cases, such as single-population species systems, single-population-sample-per-species datasets, or where the data are too weak to detect population structure).
We will instead consider these to be “candidate population” and “actual population” lineages respectively.
That is, we will provide <a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a> with the finest-grain units that could possibly be distinct populations as candidate population lineages, then use the power of the MSC model to accurately merge together our candidate populations into distinct populations (“species”, in <a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a> terminology).
For this analysis, we will err on the side of caution, not hestitating our over-split our candidate populations, as we can rely on the MSC to collapse them if there is insufficient gene flow restriction between them to form population boundaries.
As such, we will consider every distinct geographical sample to be a distinct candidate population.</p>
<table class="small-text compressed-table docutils align-default">
<colgroup>
<col style="width: 54%" />
<col style="width: 46%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Individual</p></th>
<th class="head"><p>Candidate Population Assignment</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><ul class="simple">
<li><p>L_australerasa_CA_Carson_Spur_3839</p></li>
<li><p>L_australerasa_CA_Carson_Spur_3840</p></li>
<li><p>L_australerasa_CA_Carson_Spur_3841</p></li>
</ul>
</td>
<td><p>L_australerasa_CA_Carson_Spur</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_australerasa_CA_Homewood_Canyon_5214</p></li>
</ul>
</td>
<td><p>L_australerasa_CA_Homewood_Canyon</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_australerasa_CA_Martin_Meadow_3838</p></li>
</ul>
</td>
<td><p>L_australerasa_CA_Martin_Meadow</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_australerasa_CA_Mill_Creek_5212</p></li>
<li><p>L_australerasa_CA_Mill_Creek_5213</p></li>
</ul>
</td>
<td><p>L_australerasa_CA_Mill_Creek</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_australerasa_CA_Nanny_Creek_3864</p></li>
<li><p>L_australerasa_CA_Nanny_Creek_3896</p></li>
</ul>
</td>
<td><p>L_australerasa_CA_Nanny_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_australerasa_CA_Oyster_Lake_3844</p></li>
<li><p>L_australerasa_CA_Oyster_Lake_3845</p></li>
</ul>
</td>
<td><p>L_australerasa_CA_Oyster_Lake</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_australerasa_OR_Crater_Lake_4984</p></li>
<li><p>L_australerasa_OR_Crater_Lake_4986</p></li>
</ul>
</td>
<td><p>L_australerasa_OR_Crater_Lake</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_casta_AK_Ketchikan_4894</p></li>
</ul>
</td>
<td><p>L_casta_AK_Ketchikan</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_casta_AK_Prince_of_Wales_Island_4523</p></li>
</ul>
</td>
<td><p>L_casta_AK_Prince_of_Wales_Island</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_casta_CA_Mt_Tamalpais_3830</p></li>
</ul>
</td>
<td><p>L_casta_CA_Mt_Tamalpais</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_casta_CA_Soda_Creek_4049</p></li>
</ul>
</td>
<td><p>L_casta_CA_Soda_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_casta_CA_West_Branch_Mill_Creek_3703</p></li>
</ul>
</td>
<td><p>L_casta_CA_West_Branch_Mill_Creek</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_casta_OR_Lost_Prairie_5204</p></li>
</ul>
</td>
<td><p>L_casta_OR_Lost_Prairie</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_casta_OR_Marys_Peak_2545</p></li>
</ul>
</td>
<td><p>L_casta_OR_Marys_Peak</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_casta_OR_School_Creek_3041</p></li>
</ul>
</td>
<td><p>L_casta_OR_School_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_casta_WA_Taneum_Creek_1400</p></li>
</ul>
</td>
<td><p>L_casta_WA_Taneum_Creek</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_disjuncta_BC_Summit_Creek_1896</p></li>
<li><p>L_disjuncta_BC_Summit_Creek_3090</p></li>
</ul>
</td>
<td><p>L_disjuncta_BC_Summit_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_disjuncta_CA_Emerson_Creek_4122</p></li>
</ul>
</td>
<td><p>L_disjuncta_CA_Emerson_Creek</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_disjuncta_CA_Lily_Lake_3069</p></li>
</ul>
</td>
<td><p>L_disjuncta_CA_Lily_Lake</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_disjuncta_CA_Salmon_Creek_4133</p></li>
</ul>
</td>
<td><p>L_disjuncta_CA_Salmon_Creek</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_disjuncta_CA_Trinity_Alps_4115</p></li>
</ul>
</td>
<td><p>L_disjuncta_CA_Trinity_Alps</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_disjuncta_ID_Salmon_River_4780</p></li>
</ul>
</td>
<td><p>L_disjuncta_ID_Salmon_River</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_disjuncta_MT_Mill_Creek_4716</p></li>
</ul>
</td>
<td><p>L_disjuncta_MT_Mill_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_disjuncta_OR_Lostine_River_3848</p></li>
</ul>
</td>
<td><p>L_disjuncta_OR_Lostine_River</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_disjuncta_OR_Mt_Hood_4143</p></li>
</ul>
</td>
<td><p>L_disjuncta_OR_Mt_Hood</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_erasa_AK_Thompson_Pass_4059</p></li>
</ul>
</td>
<td><p>L_erasa_AK_Thompson_Pass</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_erasa_BC_Cherryville_4002</p></li>
</ul>
</td>
<td><p>L_erasa_BC_Cherryville</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_erasa_OR_Lost_Prairie_5197</p></li>
<li><p>L_erasa_OR_Lost_Prairie_5199</p></li>
<li><p>L_erasa_OR_Lost_Prairie_5200</p></li>
<li><p>L_erasa_OR_Lost_Prairie_5201</p></li>
</ul>
</td>
<td><p>L_erasa_OR_Lost_Prairie</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_erasa_OR_Marys_Peak_2575</p></li>
<li><p>L_erasa_OR_Marys_Peak_2586</p></li>
<li><p>L_erasa_OR_Marys_Peak_2615</p></li>
<li><p>L_erasa_OR_Marys_Peak_2616</p></li>
</ul>
</td>
<td><p>L_erasa_OR_Marys_Peak</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_erasa_OR_Mount_Hebo_3013</p></li>
<li><p>L_erasa_OR_Mount_Hebo_3016</p></li>
</ul>
</td>
<td><p>L_erasa_OR_Mount_Hebo</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_erasa_OR_Mt_Hood_4144</p></li>
</ul>
</td>
<td><p>L_erasa_OR_Mt_Hood</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_erasa_OR_Prairie_Peak_2580</p></li>
</ul>
</td>
<td><p>L_erasa_OR_Prairie_Peak</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_kavanaughi_MT_Bitterroot_River_4646</p></li>
</ul>
</td>
<td><p>L_kavanaughi_MT_Bitterroot_River</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_kavanaughi_MT_Lost_Horse_Creek_4648</p></li>
</ul>
</td>
<td><p>L_kavanaughi_MT_Lost_Horse_Creek</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_kavanaughi_OR_Little_Philips_Creek_4998</p></li>
</ul>
</td>
<td><p>L_kavanaughi_OR_Little_Philips_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_kavanaughi_OR_Lostine_River_4996</p></li>
</ul>
</td>
<td><p>L_kavanaughi_OR_Lostine_River</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_kavanaughi_OR_Lostine_River_Valley_4990</p></li>
<li><p>L_kavanaughi_OR_Lostine_River_Valley_4992</p></li>
<li><p>L_kavanaughi_OR_Lostine_River_Valley_4993</p></li>
<li><p>L_kavanaughi_OR_Lostine_River_Valley_5000</p></li>
<li><p>L_kavanaughi_OR_Lostine_River_Valley_5002</p></li>
<li><p>L_kavanaughi_OR_Lostine_River_Valley_5006</p></li>
<li><p>L_kavanaughi_OR_Lostine_River_Valley_5008</p></li>
<li><p>L_kavanaughi_OR_Lostine_River_Valley_5010</p></li>
</ul>
</td>
<td><p>L_kavanaughi_OR_Lostine_River_Valley</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_lindrothi_CA_Deadman_Creek_4140</p></li>
</ul>
</td>
<td><p>L_lindrothi_CA_Deadman_Creek</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_lindrothi_CA_East_Fork_Kaweah_River_4120</p></li>
</ul>
</td>
<td><p>L_lindrothi_CA_East_Fork_Kaweah_River</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_lindrothi_CA_Emerald_Lake_4116</p></li>
<li><p>L_lindrothi_CA_Emerald_Lake_4117</p></li>
<li><p>L_lindrothi_CA_Emerald_Lake_4118</p></li>
</ul>
</td>
<td><p>L_lindrothi_CA_Emerald_Lake</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_lindrothi_CA_Kaiser_Pass_4121</p></li>
</ul>
</td>
<td><p>L_lindrothi_CA_Kaiser_Pass</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_lindrothi_CA_Long_Valley_Creek_5072</p></li>
</ul>
</td>
<td><p>L_lindrothi_CA_Long_Valley_Creek</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_lindrothi_CA_Sonora_Pass_4134</p></li>
</ul>
</td>
<td><p>L_lindrothi_CA_Sonora_Pass</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_lindrothi_CA_South_Fork_Bishop_Creek_3568</p></li>
</ul>
</td>
<td><p>L_lindrothi_CA_South_Fork_Bishop_Creek</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_lindrothi_CA_Tioga_Lake_4132</p></li>
</ul>
</td>
<td><p>L_lindrothi_CA_Tioga_Lake</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_osculans_CA_Carson_Spur_3164</p></li>
</ul>
</td>
<td><p>L_osculans_CA_Carson_Spur</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_osculans_CA_Cold_Creek_1387</p></li>
<li><p>L_osculans_CA_Cold_Creek_1390</p></li>
</ul>
</td>
<td><p>L_osculans_CA_Cold_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_osculans_CA_Los_Padres_NF_3162</p></li>
</ul>
</td>
<td><p>L_osculans_CA_Los_Padres_NF</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_osculans_CA_Nanny_Creek_3721</p></li>
</ul>
</td>
<td><p>L_osculans_CA_Nanny_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_osculans_CA_Pike_County_Gulch_3846</p></li>
</ul>
</td>
<td><p>L_osculans_CA_Pike_County_Gulch</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_osculans_CA_Rainbow_1401</p></li>
</ul>
</td>
<td><p>L_osculans_CA_Rainbow</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_osculans_CA_Stanislaus_NF_3157</p></li>
</ul>
</td>
<td><p>L_osculans_CA_Stanislaus_NF</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_osculans_CA_Strawberry_Creek_3163</p></li>
</ul>
</td>
<td><p>L_osculans_CA_Strawberry_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_osculans_CA_Warner_Range_3161</p></li>
</ul>
</td>
<td><p>L_osculans_CA_Warner_Range</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_osculans_OR_Eugene_4593</p></li>
</ul>
</td>
<td><p>L_osculans_OR_Eugene</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_osculans_OR_Goodman_Creek_3158</p></li>
</ul>
</td>
<td><p>L_osculans_OR_Goodman_Creek</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_osculans_OR_Little_Philips_Creek_5001</p></li>
</ul>
</td>
<td><p>L_osculans_OR_Little_Philips_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_osculans_OR_Middle_Fork_Berry_Creek_3095</p></li>
</ul>
</td>
<td><p>L_osculans_OR_Middle_Fork_Berry_Creek</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_osculans_OR_School_Creek_2638</p></li>
</ul>
</td>
<td><p>L_osculans_OR_School_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_osculans_OR_Walton_Lake_4743</p></li>
</ul>
</td>
<td><p>L_osculans_OR_Walton_Lake</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_probata_BC_Summit_Creek_3720</p></li>
</ul>
</td>
<td><p>L_probata_BC_Summit_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_probata_CA_Algoma_Camp_3855</p></li>
</ul>
</td>
<td><p>L_probata_CA_Algoma_Camp</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_probata_CA_Ellery_Lake_4138</p></li>
</ul>
</td>
<td><p>L_probata_CA_Ellery_Lake</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_probata_CA_Middle_Martis_Creek_1161</p></li>
<li><p>L_probata_CA_Middle_Martis_Creek_1970</p></li>
</ul>
</td>
<td><p>L_probata_CA_Middle_Martis_Creek</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_probata_CA_Nanny_Creek_3895</p></li>
</ul>
</td>
<td><p>L_probata_CA_Nanny_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_probata_CA_Sherman_Pass_3730</p></li>
</ul>
</td>
<td><p>L_probata_CA_Sherman_Pass</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_probata_CA_South_Fork_Bishop_Creek_3686</p></li>
</ul>
</td>
<td><p>L_probata_CA_South_Fork_Bishop_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_probata_CA_Squaw_Valley_Resort_5211</p></li>
</ul>
</td>
<td><p>L_probata_CA_Squaw_Valley_Resort</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_probata_CA_Strawberry_Creek_3832</p></li>
</ul>
</td>
<td><p>L_probata_CA_Strawberry_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_probata_CA_Tamarack_Lake_4137</p></li>
</ul>
</td>
<td><p>L_probata_CA_Tamarack_Lake</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_probata_CA_Warner_Range_3863</p></li>
</ul>
</td>
<td><p>L_probata_CA_Warner_Range</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_probata_CA_White_Mountains_3833</p></li>
</ul>
</td>
<td><p>L_probata_CA_White_Mountains</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_probata_ID_Baker_Creek_3865</p></li>
</ul>
</td>
<td><p>L_probata_ID_Baker_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_probata_ID_Galena_Summit_3722</p></li>
</ul>
</td>
<td><p>L_probata_ID_Galena_Summit</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_probata_ID_Park_Creek_3866</p></li>
</ul>
</td>
<td><p>L_probata_ID_Park_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_probata_MT_Mill_Creek_4713</p></li>
</ul>
</td>
<td><p>L_probata_MT_Mill_Creek</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_probata_MT_Prospect_Creek_4645</p></li>
</ul>
</td>
<td><p>L_probata_MT_Prospect_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_probata_NV_Ruby_Mountains_3684</p></li>
</ul>
</td>
<td><p>L_probata_NV_Ruby_Mountains</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_probata_OR_Little_Philips_Creek_4995</p></li>
</ul>
</td>
<td><p>L_probata_OR_Little_Philips_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_probata_OR_Lonesome_Spring_4744</p></li>
</ul>
</td>
<td><p>L_probata_OR_Lonesome_Spring</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_probata_OR_Lost_Prairie_3723</p></li>
</ul>
</td>
<td><p>L_probata_OR_Lost_Prairie</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_probata_OR_Lostine_River_Valley_4991</p></li>
</ul>
</td>
<td><p>L_probata_OR_Lostine_River_Valley</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_probata_OR_Lostine_River_Valley_5004</p></li>
</ul>
</td>
<td><p>L_probata_OR_Lostine_River_Valley</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_probata_OR_Mt_Ashland_3165</p></li>
</ul>
</td>
<td><p>L_probata_OR_Mt_Ashland</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_probata_OR_Odell_Creek_3867</p></li>
</ul>
</td>
<td><p>L_probata_OR_Odell_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_probata_OR_Steens_Mountains_2724</p></li>
<li><p>L_probata_OR_Steens_Mountains_3717</p></li>
</ul>
</td>
<td><p>L_probata_OR_Steens_Mountains</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_probata_UT_Shingle_Creek_4198</p></li>
</ul>
</td>
<td><p>L_probata_UT_Shingle_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_probata_UT_Stansbury_Mtns_3601</p></li>
<li><p>L_probata_UT_Stansbury_Mtns_3685</p></li>
</ul>
</td>
<td><p>L_probata_UT_Stansbury_Mtns</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_probata_UT_Tushar_Mountains_5037</p></li>
</ul>
</td>
<td><p>L_probata_UT_Tushar_Mountains</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_probata_WA_Blue_Mountains_3854</p></li>
</ul>
</td>
<td><p>L_probata_WA_Blue_Mountains</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_probata_WA_Taneum_Creek_1320</p></li>
</ul>
</td>
<td><p>L_probata_WA_Taneum_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_pseudoerasa_CA_Kaiser_Pass_4139</p></li>
</ul>
</td>
<td><p>L_pseudoerasa_CA_Kaiser_Pass</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_pseudoerasa_CA_Lily_Lake_3073</p></li>
</ul>
</td>
<td><p>L_pseudoerasa_CA_Lily_Lake</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_pseudoerasa_CA_Sherman_Pass_3599</p></li>
<li><p>L_pseudoerasa_CA_Sherman_Pass_3688</p></li>
</ul>
</td>
<td><p>L_pseudoerasa_CA_Sherman_Pass</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_pseudoerasa_CA_Strawberry_Creek_3072</p></li>
<li><p>L_pseudoerasa_CA_Strawberry_Creek_3083</p></li>
<li><p>L_pseudoerasa_CA_Strawberry_Creek_3086</p></li>
<li><p>L_pseudoerasa_CA_Strawberry_Creek_3087</p></li>
</ul>
</td>
<td><p>L_pseudoerasa_CA_Strawberry_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_pseudoerasa_CA_Trinity_Alps_4114</p></li>
</ul>
</td>
<td><p>L_pseudoerasa_CA_Trinity_Alps</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_sequoiae_CA_Bridal_Veil_Falls_3078</p></li>
</ul>
</td>
<td><p>L_sequoiae_CA_Bridal_Veil_Falls</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_sequoiae_CA_Nanny_Creek_3702</p></li>
</ul>
</td>
<td><p>L_sequoiae_CA_Nanny_Creek</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_sequoiae_CA_Strawberry_Creek_3075</p></li>
<li><p>L_sequoiae_CA_Strawberry_Creek_3085</p></li>
</ul>
</td>
<td><p>L_sequoiae_CA_Strawberry_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_sequoiae_OR_Oakridge_2647</p></li>
</ul>
</td>
<td><p>L_sequoiae_OR_Oakridge</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_sequoiae_OR_School_Creek_2614</p></li>
</ul>
</td>
<td><p>L_sequoiae_OR_School_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_tuulukwa_CA_Trinity_Alps_4113</p></li>
</ul>
</td>
<td><p>L_tuulukwa_CA_Trinity_Alps</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_tuulukwa_OR_Knowles_Creek_3700</p></li>
<li><p>L_tuulukwa_OR_Knowles_Creek_3701</p></li>
</ul>
</td>
<td><p>L_tuulukwa_OR_Knowles_Creek</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_tuulukwa_OR_Marys_Peak_2581</p></li>
<li><p>L_tuulukwa_OR_Marys_Peak_2635</p></li>
<li><p>L_tuulukwa_OR_Marys_Peak_2636</p></li>
<li><p>L_tuulukwa_OR_Marys_Peak_2637</p></li>
<li><p>L_tuulukwa_OR_Marys_Peak_2642</p></li>
<li><p>L_tuulukwa_OR_Marys_Peak_2643</p></li>
<li><p>L_tuulukwa_OR_Marys_Peak_3782</p></li>
</ul>
</td>
<td><p>L_tuulukwa_OR_Marys_Peak</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="generating-a-guide-tree-for-population-delimitation">
<h3>Generating a Guide Tree for Population Delimitation<a class="headerlink" href="#generating-a-guide-tree-for-population-delimitation" title="Permalink to this headline">¶</a></h3>
<p>We will provide <a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a> with a guide tree for its population delimitation analysis.
We will use <a class="reference external" href="https://taming-the-beast.org/tutorials/starbeast2-tutorial/"><span class="program">StarBeast2</span></a> to generate this guide tree.</p>
<p>The full <a class="reference external" href="https://taming-the-beast.org/tutorials/starbeast2-tutorial/"><span class="program">StarBeast2</span></a> configuration file, generated using <code class="docutils literal notranslate"><span class="pre">BEAUTi</span></code>, can be found at in <span class="filepath">lionepha/01-guidetree-estimation/sb00500M.xml</span>.
The alignments listed above were imported, and the following “traits” file was used to map alignment sequences to canidate population units: <span class="filepath">lionepha/01-guidetree-estimation/traits.txt</span>.</p>
<p>We used a single strict clock model across all genes, and a HKY+G model of substitution.</p>
<p>We ran four replicates of this analysis for 500 million generations each, sampling every 500,000 generations for a total of 1000 samples from each replicates.
The first 250 samples were discarded from the 100 samples of each replicates.
Convergence was diagnosed through inspection of traces for each parameter as well as the likelihood and posterior.
ESS values for each parameter were established to be more than 250, and distributions of parameter values were compared to a “null” run (i.e., a run without data where just the prior was sampled) to confirm that the analysis learned from the data sufficiently to shift the posterior away from the prior.</p>
<p>The post-burn in samples from the posterior were summarized using <a class="reference external" href="https://dendropy.org/programs/sumtrees.html"><span class="program">SumTrees</span></a>, with the following command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ sumtrees.py -b 250 \
            -s mcct \
            -e clear \
            -l clear \
            --force-rooted \
            --suppress-annotations \
            -r -o summary.guide.tre \
            run1/species.trees \
            run2/species.trees \
            run3/species.trees \
            run4/species.trees
</pre></div>
</div>
<p>This selects the Maximum Clade Credibility Tree (MCCT) tree for the summary topolgy, stripping all branch lengths and metadata annotations, to result in the following:</p>
<p class="framebox center"><span class="filepath">lionepha/01-guidetree-estimation/guidetree.nex</span></p>
</div>
<div class="section" id="delimitation-of-population-units">
<h3>Delimitation of Population Units<a class="headerlink" href="#delimitation-of-population-units" title="Permalink to this headline">¶</a></h3>
<p>Now that we have a guide tree that treats each distinct geographical lineage as a candidate distinct Wright-Fisher population, we will run <a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a> in “A10” mode to delimit the true population units under the “multipopulation coalescent” (i.e., the MSC).</p>
<div class="section" id="for-small-datasets-the-single-analysis-approach">
<h4>For Small Datasets: the Single-Analysis Approach<a class="headerlink" href="#for-small-datasets-the-single-analysis-approach" title="Permalink to this headline">¶</a></h4>
<p>The files provided in the <span class="filepath">lionepha/02a-population-delimitation-pooled</span> directory set up a fairly straightforward <a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a> “A10” analyses using the data that we have collected and the guide tree we have estimated:</p>
<table class="colwidths-given compressed-table docutils align-default" id="id1">
<caption><span class="caption-text">Pooled <a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a> Population Delimitation Analysis</span><a class="headerlink" href="#id1" title="Permalink to this table">¶</a></caption>
<colgroup>
<col style="width: 50%" />
<col style="width: 50%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>File</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>bpprun.input.ctl</p></td>
<td><p>Control file</p></td>
</tr>
<tr class="row-odd"><td><p>bpprun.input.chars.txt</p></td>
<td><p>Character data</p></td>
</tr>
<tr class="row-even"><td><p>bpprun.input.imap.txt</p></td>
<td><p>Sequence to population map</p></td>
</tr>
<tr class="row-odd"><td><p>bpp00.job</p></td>
<td><p>Execution job</p></td>
</tr>
</tbody>
</table>
<p>The problem is that, when executing the analysis by either running the job file:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ bash bpp00.job
</pre></div>
</div>
<p>or submitting it to an execution host:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ qsub bpp00.job
</pre></div>
</div>
<p>or simply running the command directly:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ bpp --cfile bpprun.input.ctl
</pre></div>
</div>
<p>we might find that the data set is too large to analyze:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">bpp</span> <span class="n">v4</span><span class="o">.</span><span class="mf">1.4</span><span class="n">_linux_x86_64</span><span class="p">,</span> <span class="mi">31</span><span class="n">GB</span> <span class="n">RAM</span><span class="p">,</span> <span class="mi">12</span> <span class="n">cores</span>
<span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">bpp</span>
<span class="o">.</span>
<span class="o">.</span>
<span class="o">.</span>
<span class="n">Total</span> <span class="n">species</span> <span class="n">delimitations</span><span class="p">:</span> <span class="mi">23958541050464777</span>
<span class="n">Unable</span> <span class="n">to</span> <span class="n">allocate</span> <span class="n">enough</span> <span class="n">memory</span><span class="o">.</span>
</pre></div>
</div>
</div>
<div class="section" id="for-larger-datasets-the-subtree-decomposition-approach">
<h4>For Larger Datasets: the Subtree Decomposition Approach<a class="headerlink" href="#for-larger-datasets-the-subtree-decomposition-approach" title="Permalink to this headline">¶</a></h4>
<p>The solution is to break the data set up into subtrees and carry out a separate population delimitation analysis on each subtree.
While designing a decomposition scheme, it is important to note that we should <em>not</em> separate out candidate population lineages that might potentially belong to the same (actual) population into separate subtrees, thereby preventing <a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a> from being able to collapse them if it does not detect any gene flow restriction between them.
In most studies, this should not be too difficult to identify.
Even in cases where we might not know where the <em>species</em> boundaries are, with reference to the guide tree phylogeny we should be able to identify subtree clusters that do not disrupt population boundaries.
In this case of this <em>Lionepha</em> study, the nominal species clades provide very nice granularity — small enough to be analyzed by <a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a>, yet with no danger of breaking up an actual population.</p>
<div class="figure align-default">
<a class="figure-image reference internal image-reference" href="_images/lionepha-guidetree.png"><img alt="Guide tree for Lionepha population delimitation, showing subtrees used when decomposing into separate analyses." class="figure-image" src="_images/lionepha-guidetree.png" style="width: 100%;" /></a>
</div>
<p class="figure-caption"><strong>Figure</strong>:  We cannot delimit populations for the entire data set simulatneously in <a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a> due to the number of lineages. Instead, we decompose the analysis into a set of smaller analyses based on subtrees.</p>
<p>The set up for this set of analyses can be found at:</p>
<p class="framebox center"><span class="filepath">lionepha/02b-population-delimitation-subtrees</span></p>
<p>with each subdirectory containing a stand-alone analysis.</p>
</div>
<div class="section" id="collating-results-of-the-subtree-approach">
<h4>Collating Results of the Subtree Approach<a class="headerlink" href="#collating-results-of-the-subtree-approach" title="Permalink to this headline">¶</a></h4>
<p>Each of the subtree analysis now has the populations delimited under the MSC model.
Having identified these population units across various subtrees, we now need to collate and pool them.
DELINEATE helpfully provides a script for you to do this fairly robustly: <a class="reference external" href="https://delineate.org/"><span class="program">delineate-bppsum</span></a>.
This script takes as its input two sets of files:</p>
<ul class="simple">
<li><p>the “imap” files you provided to <a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a> as input, which maps sequences to candidate population lineages</p></li>
<li><p>the output log of <a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a> analyses (<em>not</em> the MCMC log), which provides a tree at the end with posterior probability of internal nodes indicated by labels</p></li>
</ul>
<p>These files can be specified in any order, but must collectively span the entire analysis: the set of all candidate population names defined across all “imap” files must be equal to the set of all candidate population names found across all trees in all <a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a> output log files.</p>
<p>In this example, we have all the independent subtree analyses packed away in subdirectories, “<code class="docutils literal notranslate"><span class="pre">01</span></code>”, “<code class="docutils literal notranslate"><span class="pre">02</span></code>”, etc.
Assuming we are in the <a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a> analysis subdirectory, <span class="filepath">lionepha/02b-population-delimitation-subtrees</span>, we could just type in all the paths:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span> \
    <span class="o">--</span><span class="n">imap</span>    <span class="mi">01</span><span class="o">/</span><span class="n">bpprun</span><span class="o">.</span><span class="n">input</span><span class="o">.</span><span class="n">imap</span><span class="o">.</span><span class="n">txt</span> \
              <span class="mi">02</span><span class="o">/</span><span class="n">bpprun</span><span class="o">.</span><span class="n">input</span><span class="o">.</span><span class="n">imap</span><span class="o">.</span><span class="n">txt</span> \
              <span class="o">.</span>
              <span class="o">.</span>
              <span class="p">(</span><span class="n">etc</span><span class="o">.</span><span class="p">)</span>
    <span class="o">--</span><span class="n">results</span> <span class="mi">01</span><span class="o">/</span><span class="n">results</span><span class="o">.</span><span class="n">out</span><span class="o">.</span><span class="n">txt</span> \
              <span class="mi">02</span><span class="o">/</span><span class="n">results</span><span class="o">.</span><span class="n">out</span><span class="o">.</span><span class="n">txt</span> \
              <span class="o">.</span>
              <span class="o">.</span>
              <span class="p">(</span><span class="n">etc</span><span class="o">.</span><span class="p">)</span>
</pre></div>
</div>
<p>but because of judicious naming of the files, we can use some basic shell commands to generate the list of input files:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>delineate-bppsum \
    --imap $(find . -name &quot;*imap*&quot;) \
    --results $(find . -name &quot;*out.txt&quot;)
</pre></div>
</div>
<p>Executing the above command results in:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="mi">11</span> <span class="n">BPP</span> <span class="s1">&#39;imap&#39;</span> <span class="n">files</span> <span class="n">specified</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">mapping</span> <span class="n">file</span>   <span class="mi">1</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">00</span><span class="o">/</span><span class="n">bpprun</span><span class="o">.</span><span class="n">input</span><span class="o">.</span><span class="n">imap</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">13</span> <span class="n">lineages</span><span class="p">,</span> <span class="mi">7</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">mapping</span> <span class="n">file</span>   <span class="mi">2</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">01</span><span class="o">/</span><span class="n">bpprun</span><span class="o">.</span><span class="n">input</span><span class="o">.</span><span class="n">imap</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">9</span> <span class="n">lineages</span><span class="p">,</span> <span class="mi">9</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">mapping</span> <span class="n">file</span>   <span class="mi">3</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">02</span><span class="o">/</span><span class="n">bpprun</span><span class="o">.</span><span class="n">input</span><span class="o">.</span><span class="n">imap</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">10</span> <span class="n">lineages</span><span class="p">,</span> <span class="mi">9</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">mapping</span> <span class="n">file</span>   <span class="mi">4</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">03</span><span class="o">/</span><span class="n">bpprun</span><span class="o">.</span><span class="n">input</span><span class="o">.</span><span class="n">imap</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">14</span> <span class="n">lineages</span><span class="p">,</span> <span class="mi">7</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">mapping</span> <span class="n">file</span>   <span class="mi">5</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">04</span><span class="o">/</span><span class="n">bpprun</span><span class="o">.</span><span class="n">input</span><span class="o">.</span><span class="n">imap</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">12</span> <span class="n">lineages</span><span class="p">,</span> <span class="mi">5</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">mapping</span> <span class="n">file</span>   <span class="mi">6</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">05</span><span class="o">/</span><span class="n">bpprun</span><span class="o">.</span><span class="n">input</span><span class="o">.</span><span class="n">imap</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">10</span> <span class="n">lineages</span><span class="p">,</span> <span class="mi">8</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">mapping</span> <span class="n">file</span>   <span class="mi">7</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">06</span><span class="o">/</span><span class="n">bpprun</span><span class="o">.</span><span class="n">input</span><span class="o">.</span><span class="n">imap</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">16</span> <span class="n">lineages</span><span class="p">,</span> <span class="mi">15</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">mapping</span> <span class="n">file</span>   <span class="mi">8</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">07</span><span class="o">/</span><span class="n">bpprun</span><span class="o">.</span><span class="n">input</span><span class="o">.</span><span class="n">imap</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">34</span> <span class="n">lineages</span><span class="p">,</span> <span class="mi">30</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">mapping</span> <span class="n">file</span>   <span class="mi">9</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">08</span><span class="o">/</span><span class="n">bpprun</span><span class="o">.</span><span class="n">input</span><span class="o">.</span><span class="n">imap</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">9</span> <span class="n">lineages</span><span class="p">,</span> <span class="mi">5</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">mapping</span> <span class="n">file</span>  <span class="mi">10</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">09</span><span class="o">/</span><span class="n">bpprun</span><span class="o">.</span><span class="n">input</span><span class="o">.</span><span class="n">imap</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">6</span> <span class="n">lineages</span><span class="p">,</span> <span class="mi">5</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">mapping</span> <span class="n">file</span>  <span class="mi">11</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">10</span><span class="o">/</span><span class="n">bpprun</span><span class="o">.</span><span class="n">input</span><span class="o">.</span><span class="n">imap</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">10</span> <span class="n">lineages</span><span class="p">,</span> <span class="mi">3</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="mi">11</span> <span class="n">BPP</span> <span class="n">output</span> <span class="n">files</span> <span class="n">specified</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">output</span> <span class="n">file</span>   <span class="mi">1</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">00</span><span class="o">/</span><span class="n">results</span><span class="o">.</span><span class="n">out</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">7</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">output</span> <span class="n">file</span>   <span class="mi">2</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">01</span><span class="o">/</span><span class="n">results</span><span class="o">.</span><span class="n">out</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">9</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">output</span> <span class="n">file</span>   <span class="mi">3</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">02</span><span class="o">/</span><span class="n">results</span><span class="o">.</span><span class="n">out</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">9</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">output</span> <span class="n">file</span>   <span class="mi">4</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">03</span><span class="o">/</span><span class="n">results</span><span class="o">.</span><span class="n">out</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">7</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">output</span> <span class="n">file</span>   <span class="mi">5</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">04</span><span class="o">/</span><span class="n">results</span><span class="o">.</span><span class="n">out</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">5</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">output</span> <span class="n">file</span>   <span class="mi">6</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">05</span><span class="o">/</span><span class="n">results</span><span class="o">.</span><span class="n">out</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">8</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">output</span> <span class="n">file</span>   <span class="mi">7</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">06</span><span class="o">/</span><span class="n">results</span><span class="o">.</span><span class="n">out</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">15</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">output</span> <span class="n">file</span>   <span class="mi">8</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">07</span><span class="o">/</span><span class="n">results</span><span class="o">.</span><span class="n">out</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">30</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">output</span> <span class="n">file</span>   <span class="mi">9</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">08</span><span class="o">/</span><span class="n">results</span><span class="o">.</span><span class="n">out</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">5</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">output</span> <span class="n">file</span>  <span class="mi">10</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">09</span><span class="o">/</span><span class="n">results</span><span class="o">.</span><span class="n">out</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">5</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="o">-</span> <span class="n">Reading</span> <span class="n">output</span> <span class="n">file</span>  <span class="mi">11</span> <span class="n">of</span> <span class="mi">11</span><span class="p">:</span> <span class="o">./</span><span class="mi">10</span><span class="o">/</span><span class="n">results</span><span class="o">.</span><span class="n">out</span><span class="o">.</span><span class="n">txt</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span>   <span class="o">-</span> <span class="p">(</span><span class="mi">3</span> <span class="n">candidate</span> <span class="n">populations</span><span class="p">)</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="n">Posterior</span> <span class="n">probability</span> <span class="n">threshold</span> <span class="n">of</span> <span class="mf">0.50</span><span class="p">:</span> <span class="mi">86</span> <span class="n">populations</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="n">Posterior</span> <span class="n">probability</span> <span class="n">threshold</span> <span class="n">of</span> <span class="mf">0.75</span><span class="p">:</span> <span class="mi">72</span> <span class="n">populations</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="n">Posterior</span> <span class="n">probability</span> <span class="n">threshold</span> <span class="n">of</span> <span class="mf">0.90</span><span class="p">:</span> <span class="mi">65</span> <span class="n">populations</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="n">Posterior</span> <span class="n">probability</span> <span class="n">threshold</span> <span class="n">of</span> <span class="mf">0.95</span><span class="p">:</span> <span class="mi">59</span> <span class="n">populations</span>
<span class="p">[</span><span class="n">delineate</span><span class="o">-</span><span class="n">bppsum</span><span class="p">]</span> <span class="n">Posterior</span> <span class="n">probability</span> <span class="n">threshold</span> <span class="n">of</span> <span class="mf">1.00</span><span class="p">:</span> <span class="mi">46</span> <span class="n">populations</span>
</pre></div>
</div>
<p>and produces the following files:</p>
<ul class="simple">
<li><p><span class="filepath">coalescent-pops.sb2-traits.p050.txt</span></p></li>
<li><p><span class="filepath">coalescent-pops.sb2-traits.p075.txt</span></p></li>
<li><p><span class="filepath">coalescent-pops.sb2-traits.p090.txt</span></p></li>
<li><p><span class="filepath">coalescent-pops.sb2-traits.p095.txt</span></p></li>
<li><p><span class="filepath">coalescent-pops.sb2-traits.p100.txt</span></p></li>
<li><p><span class="filepath">coalescent-pops.summary.csv</span></p></li>
</ul>
<p>The population boundaries and identities of the various individuals are reported at different posterior probabilities (0.50, 0.75, 0.90, 0.95, and 1.00).
A comprehensive overview of all the identities under the different posterior probabilities is provided in the file: <span class="filepath">coalescent-pops.summary.csv</span>.
The other files (ending with filenames “…sb2-traits.p0xx.txt”) are <a class="reference external" href="https://taming-the-beast.org/tutorials/starbeast2-tutorial/"><span class="program">StarBeast2</span></a> “<code class="docutils literal notranslate"><span class="pre">traits</span></code>” files at each of those posterior probability thresholds.
These latter make setting up a <a class="reference external" href="https://taming-the-beast.org/tutorials/starbeast2-tutorial/"><span class="program">StarBeast2</span></a> analysis to estimate an ultrametric phylogeny relating the population units delimited at each of the posterior probability thresholds very straightforward.
With the latter, do <em>not</em> be confused by the column header “species”!
Again, this is simply due to the misleading terminology adopted for the higher-level units of organization in <a class="reference external" href="https://taming-the-beast.org/tutorials/starbeast2-tutorial/"><span class="program">StarBeast2</span></a> and most other programs that use the MSC.
What we are working with here are <em>population units</em>, some of which <em>may</em> also correspond to species while others may not.
This is abundantly clear by the fact that the 12 nominal species — established with both genomic as well as morphological data contributing toward an understanding of the system over a <em>decade</em> of the study — is split into 46 to 86 units (!) by the MSC analysis, depending on the posterior probability threshold we adopt for those units.
Interpreting these units as “species” is categorically and unconditionally wrong, and would represent a jettisoning of a basic understanding of speciation, systematics, and biology, due to a complete misunderstanding of the MSC model and misinterpretation of its results.</p>
<p class="emphatic">The MSC delimits <em>populations</em>, not species.</p>
<p>Note that where our candidate populations assignment have indeed turned out to be distinct population lineages under a particular posterior probability threshold, the original population label is retained (e.g., “L_australerasa_CA_Martin_Meadow” or “L_probata_WA_Blue_Mountains”).</p>
<p>In cases where multiple candidate population lineages have been “collapsed”, because there was insufficient signal for restriction of Wright-Fisher panmixia between them detected at a particular posterior probability threshold, they have been relabeled with synthetic labels (e.g. “coalescentpop001”, “coalescentpop002”, etc.).
Thus, for example, we see that, in the setup to our <a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a> analysies, we allowed for the possibility that  <em>L. erasa</em> from Mary’s Peak, Mount Hebo, Mount Hood, and Prairie Peak might each be a distinct population:</p>
<table class="small-text compressed-table center docutils align-default">
<colgroup>
<col style="width: 43%" />
<col style="width: 57%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>sample</p></th>
<th class="head"><p>BP&amp;B Candidate Population (“Species”)</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>L_erasa_OR_Marys_Peak_2575</p></td>
<td><p>L_erasa_OR_Marys_Peak</p></td>
</tr>
<tr class="row-odd"><td><p>L_erasa_OR_Marys_Peak_2586</p></td>
<td><p>L_erasa_OR_Marys_Peak</p></td>
</tr>
<tr class="row-even"><td><p>L_erasa_OR_Marys_Peak_2615</p></td>
<td><p>L_erasa_OR_Marys_Peak</p></td>
</tr>
<tr class="row-odd"><td><p>L_erasa_OR_Marys_Peak_2616</p></td>
<td><p>L_erasa_OR_Marys_Peak</p></td>
</tr>
<tr class="row-even"><td><p>L_erasa_OR_Mount_Hebo_3013</p></td>
<td><p>L_erasa_OR_Mount_Hebo</p></td>
</tr>
<tr class="row-odd"><td><p>L_erasa_OR_Mount_Hebo_3016</p></td>
<td><p>L_erasa_OR_Mount_Hebo</p></td>
</tr>
<tr class="row-even"><td><p>L_erasa_OR_Mt_Hood_4144</p></td>
<td><p>L_erasa_OR_Mt_Hood</p></td>
</tr>
<tr class="row-odd"><td><p>L_erasa_OR_Prairie_Peak_2580</p></td>
<td><p>L_erasa_OR_Prairie_Peak</p></td>
</tr>
</tbody>
</table>
<p>Following the analysis, we see that at a 0.95 posterior probability threshold (see the “p095” column in <span class="filepath">02b-population-delimitation-subtrees/coalescent-pops.summary.csv</span> or <span class="filepath">02b-population-delimitation-subtrees/coalescent-pops.sb2-traits.p095.txt</span>) there was insufficient evidence to support four distinct populations here.
Population boundaries between Marys Peak, Mount Hebo, and Prairie Peak was were not strong, and all individuals here were assigned to a single merged or collapsed population, “coalescentpop007”.
Only the Mount Hood individuals were separated out to be delimited as a distinct population.</p>
<table class="small-text compressed-table center docutils align-default">
<colgroup>
<col style="width: 60%" />
<col style="width: 40%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>sample</p></th>
<th class="head"><p>p095</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>L_erasa_OR_Marys_Peak_2575</p></td>
<td><p>coalescentpop007</p></td>
</tr>
<tr class="row-odd"><td><p>L_erasa_OR_Marys_Peak_2586</p></td>
<td><p>coalescentpop007</p></td>
</tr>
<tr class="row-even"><td><p>L_erasa_OR_Marys_Peak_2615</p></td>
<td><p>coalescentpop007</p></td>
</tr>
<tr class="row-odd"><td><p>L_erasa_OR_Marys_Peak_2616</p></td>
<td><p>coalescentpop007</p></td>
</tr>
<tr class="row-even"><td><p>L_erasa_OR_Mount_Hebo_3013</p></td>
<td><p>coalescentpop007</p></td>
</tr>
<tr class="row-odd"><td><p>L_erasa_OR_Mount_Hebo_3016</p></td>
<td><p>coalescentpop007</p></td>
</tr>
<tr class="row-even"><td><p>L_erasa_OR_Mt_Hood_4144</p></td>
<td><p>L_erasa_OR_Mt_Hood</p></td>
</tr>
<tr class="row-odd"><td><p>L_erasa_OR_Prairie_Peak_2580</p></td>
<td><p>coalescentpop007</p></td>
</tr>
</tbody>
</table>
<p>On the other hand, however, at the 0.50 posterior probability threshold, while the Mary’s Peak and Prairie Peak candidate population lineages were still collapsed into a single population (“coalescentpop004”), the Mount Hebo and Mount Hood individuals were each estimated to form a distinct population unto themselves (see the “p050” column in <span class="filepath">02b-population-delimitation-subtrees/coalescent-pops.summary.csv</span> or <span class="filepath">02b-population-delimitation-subtrees/coalescent-pops.sb2-traits.p050.txt</span>):</p>
<table class="small-text compressed-table center docutils align-default">
<colgroup>
<col style="width: 57%" />
<col style="width: 43%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>sample</p></th>
<th class="head"><p>p050</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>L_erasa_OR_Marys_Peak_2575</p></td>
<td><p>coalescentpop004</p></td>
</tr>
<tr class="row-odd"><td><p>L_erasa_OR_Marys_Peak_2586</p></td>
<td><p>coalescentpop004</p></td>
</tr>
<tr class="row-even"><td><p>L_erasa_OR_Marys_Peak_2615</p></td>
<td><p>coalescentpop004</p></td>
</tr>
<tr class="row-odd"><td><p>L_erasa_OR_Marys_Peak_2616</p></td>
<td><p>coalescentpop004</p></td>
</tr>
<tr class="row-even"><td><p>L_erasa_OR_Mount_Hebo_3013</p></td>
<td><p>L_erasa_OR_Mount_Hebo</p></td>
</tr>
<tr class="row-odd"><td><p>L_erasa_OR_Mount_Hebo_3016</p></td>
<td><p>L_erasa_OR_Mount_Hebo</p></td>
</tr>
<tr class="row-even"><td><p>L_erasa_OR_Mt_Hood_4144</p></td>
<td><p>L_erasa_OR_Mt_Hood</p></td>
</tr>
<tr class="row-odd"><td><p>L_erasa_OR_Prairie_Peak_2580</p></td>
<td><p>coalescentpop004</p></td>
</tr>
</tbody>
</table>
<p>Going forward to the next stage of analysis, we have to decide the posterior probability threshold at which we want to determine our population units.
For this example, we will (somewhat arbitrarily) decide on 0.95.</p>
</div>
</div>
</div>
<div class="section" id="stage-ii-generating-the-multipopulation-coalescent-ultrametric-phylogeny-of-populations">
<h2>Stage II. Generating the (Multipopulation Coalescent, Ultrametric) Phylogeny of Populations<a class="headerlink" href="#stage-ii-generating-the-multipopulation-coalescent-ultrametric-phylogeny-of-populations" title="Permalink to this headline">¶</a></h2>
<p>We use <a class="reference external" href="https://taming-the-beast.org/tutorials/starbeast2-tutorial/"><span class="program">StarBeast2</span></a> to estimate an ultrametric phylogeny of population lineages.
We use the original set of alignments (found in <span class="filepath">lionepha/00-alignments</span>) for as the input data for this, and a “<code class="docutils literal notranslate"><span class="pre">traits</span></code>” file that maps each of the sequence labels in the alignment to population identities assigned in the previous step.
For this exercise, we shall use population units that were delimited with 0.95 posterior probability, and the “<code class="docutils literal notranslate"><span class="pre">traits</span></code>” file corresponding to this is given by <span class="filepath">coalescent-pops.sb2-traits.p095.txt</span>.
The analysis setup for the <span class="program">StarBeast2</span> run can be found in the <span class="filepath">lionepha/03-population-tree-estimation/</span> directory.
There are six <span class="program">StarBeast2</span> XML configuration files.
These all set up the same “species tree” analysis, the only difference being the name prefixes of the output files.
This is so that we can run multiple independent analyses, the <em>sin qua non</em> of MCMC best practices.</p>
<p>Again, we used a single strict clock model across all genes, and a HKY+G model of substitution.
Each analysis was run for 400 million generations, with a sampling frequency of 400,000 generations.
Thus, we obtained 1000 samples from the posterior from each of the six replicates.
We can again use <a class="reference external" href="https://dendropy.org/programs/sumtrees.html"><span class="program">SumTrees</span></a> to summarize the results with the following command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ sumtrees.py \
    -b 250 \
    -s mcct \
    -e mean-age \
    -l clear \
    --force-rooted \
    --suppress-annotations \
    -ro lionepha-p095-hkyg.mcct-mean-age.tree \
    *.species.trees
</pre></div>
</div>
<p>This will discard the first 250 of the 1000 samples from each replicate as burn-in, merge the results, and select the Maximum Clade Credibility Tree (MCCT) as the summary topology, with branch lengths set such that the internal node ages are the mean of the the corresponding node ages across all post-burnin samples.
The result of this can be found here: <span class="filepath">lionepha/03-population-tree-estimation/lionepha-p095-hkyg.mcct-mean-age.tree.nex</span>.
This population-level phylogeny will be one of the two primary inputs to DELINEATE.</p>
</div>
<div class="section" id="stage-iii-assignment-of-known-vs-unknown-species-identities">
<h2>Stage III. Assignment of Known vs. Unknown Species Identities<a class="headerlink" href="#stage-iii-assignment-of-known-vs-unknown-species-identities" title="Permalink to this headline">¶</a></h2>
<p>We now inspect our phylogeny and determine <em>a priori</em> species assignments for as many population lineages as we can.
These assignments will be made with reference to our integrative understanding of the system in conjunction with the evolutionary relationships between population lineages shown in the phylogeny.</p>
<p>There will be some population lineages for which the species identity is uncontroversial — i.e., they can be unambiguously assigned to known species (or even new ones) based on morphological evidence of individuals in that population.
In the case of the “collapsed” populations (labeled “coalescentpop001”, “coalescentpop002”, etc.), we would need to examine all individuals in those units, and if <em>any</em> one of them can be definitely assigned to an independent species status based on systematic evidence (i.e., distinct species from all others in the system), then the <em>entire</em> population lineage would get assigned to a distinct species.
The reasoning behind this is that, based on previous stages of the analysis using <a class="reference external" href="https://github.com/bpp/bpp"><span class="program">BP&amp;P</span></a>, we have already decided that all individuals in that population constitute a single cohesive population, so the species identity of any one individual in that lineage would necessarily be shared by all other individuals in the same lineage.</p>
<p>There will, of course, be population units for which we will <em>not</em> be able to confidentally assign to a distinct species status — either a known, existing one, or to a new one.
These population lineages <em>could</em> be distinct species unto themselves, or could be populations of other species in the system (named or otherwise).
These are the actual targets of delimitation — we would to determine whether or not the boundaries between them and other lineages are species boundaries or population boundaries.</p>
<p>The first set of population lineages — the ones for which we can confidently assign species identities, and, hence, boundaries — constitute our “constrained” lineages.
We set them as constraints when we configure the DELINEATE analysis by setting their status to “1” in the constraints table.
This can be seen with, for e.g., “coalescentpop001”, “coalescentpop002”, and “L_australerasa_CA_Martin_Meadow” in the example constraints file (<span class="filepath">04-species-delimitation/lionepha.run1.tsv</span>; see below), which all have been assigned to “<em>australerasa</em>” (as can be seen in the “species” column), with the “status” set to 1.
This will indicate to DELINEATE that the assignment of these populations to species is fixed, and we should not consider any partition that violates this.</p>
<table class="small-text compressed-table center docutils align-default">
<colgroup>
<col style="width: 66%" />
<col style="width: 22%" />
<col style="width: 13%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>lineage</p></th>
<th class="head"><p>species</p></th>
<th class="head"><p>status</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><ul class="simple">
<li><p>coalescentpop001</p></li>
</ul>
</td>
<td><p>australerasa</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>coalescentpop002</p></li>
</ul>
</td>
<td><p>australerasa</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_australerasa_CA_Martin_Meadow</p></li>
</ul>
</td>
<td><p>australerasa</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_casta_CA_West_Branch_Mill_Creek</p></li>
</ul>
</td>
<td><p>casta</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>coalescentpop005</p></li>
</ul>
</td>
<td><p>casta</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>coalescentpop003</p></li>
</ul>
</td>
<td><p>casta</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>coalescentpop004</p></li>
</ul>
</td>
<td><p>casta</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_casta_OR_Lost_Prairie</p></li>
</ul>
</td>
<td><p>casta</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_disjuncta_OR_Lostine_River</p></li>
</ul>
</td>
<td><p>disjuncta</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_disjuncta_BC_Summit_Creek</p></li>
</ul>
</td>
<td><p>disjuncta</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_disjuncta_CA_Emerson_Creek</p></li>
</ul>
</td>
<td><p>disjuncta</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_disjuncta_MT_Mill_Creek</p></li>
</ul>
</td>
<td><p>disjuncta</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_disjuncta_ID_Salmon_River</p></li>
</ul>
</td>
<td><p>disjuncta</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_erasa_BC_Cherryville</p></li>
</ul>
</td>
<td><p>erasa</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_erasa_OR_Lost_Prairie</p></li>
</ul>
</td>
<td><p>erasa</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_erasa_OR_Mt_Hood</p></li>
</ul>
</td>
<td><p>erasa</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>coalescentpop007</p></li>
</ul>
</td>
<td><p>erasa</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_erasa_AK_Thompson_Pass</p></li>
</ul>
</td>
<td><p>erasa</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_lindrothi_CA_Long_Valley_Creek</p></li>
</ul>
</td>
<td><p>lindrothi</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_lindrothi_CA_Tioga_Lake</p></li>
</ul>
</td>
<td><p>lindrothi</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_osculans_CA_Cold_Creek</p></li>
</ul>
</td>
<td><p>osculans</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>coalescentpop015</p></li>
</ul>
</td>
<td><p>osculans</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>coalescentpop014</p></li>
</ul>
</td>
<td><p>osculans</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>coalescentpop013</p></li>
</ul>
</td>
<td><p>osculans</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>coalescentpop012</p></li>
</ul>
</td>
<td><p>osculans</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_osculans_OR_Little_Philips_Creek</p></li>
</ul>
</td>
<td><p>osculans</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_osculans_CA_Strawberry_Creek</p></li>
</ul>
</td>
<td><p>osculans</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_probata_MT_Mill_Creek</p></li>
</ul>
</td>
<td><p>probata</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>coalescentpop023</p></li>
</ul>
</td>
<td><p>probata</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_probata_CA_Warner_Range</p></li>
</ul>
</td>
<td><p>probata</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_probata_OR_Lostine_River_Valley</p></li>
</ul>
</td>
<td><p>probata</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_probata_CA_Ellery_Lake</p></li>
</ul>
</td>
<td><p>probata</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_probata_OR_Mt_Ashland</p></li>
</ul>
</td>
<td><p>probata</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_probata_WA_Blue_Mountains</p></li>
</ul>
</td>
<td><p>probata</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>coalescentpop019</p></li>
</ul>
</td>
<td><p>probata</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>coalescentpop020</p></li>
</ul>
</td>
<td><p>probata</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>coalescentpop021</p></li>
</ul>
</td>
<td><p>probata</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>coalescentpop022</p></li>
</ul>
</td>
<td><p>probata</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>coalescentpop016</p></li>
</ul>
</td>
<td><p>probata</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>coalescentpop017</p></li>
</ul>
</td>
<td><p>probata</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>coalescentpop018</p></li>
</ul>
</td>
<td><p>probata</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_pseudoerasa_CA_Kaiser_Pass</p></li>
</ul>
</td>
<td><p>pseudoerasa</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>coalescentpop025</p></li>
</ul>
</td>
<td><p>pseudoerasa</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>coalescentpop026</p></li>
</ul>
</td>
<td><p>sequoiae</p></td>
<td><p>1</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>coalescentpop027</p></li>
</ul>
</td>
<td><p>sequoiae</p></td>
<td><p>1</p></td>
</tr>
</tbody>
</table>
<p>Conversely, there remain a number of population lineages that we are <em>not</em> sure about (we are going to pretend, for the purposes of this exercise).
These populations <em>could</em> be populations of existing species <em>OR</em> they could be independent species (either as a single lineage or clusters of lineages).
These are the lineages that are the (remaining) primary focus of the species delimitation analysis.
In the constraints table, we set the status of these lineages to “0”, indicating that we want the DELINEATE analysis to explore all possible partitions that vary in the species assignments  of these populations.
Note that we still have entries in the “species” field for this “unconstrained” lineages.
This is purely for our own book-keeping and will be ignored entirely by DELINEATE.
We could, in principle, replace them with any text or level them blank, and they will have no difference in the program running or output.</p>
<table class="small-text compressed-table center docutils align-default">
<colgroup>
<col style="width: 66%" />
<col style="width: 22%" />
<col style="width: 13%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>lineage</p></th>
<th class="head"><p>species</p></th>
<th class="head"><p>status</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><ul class="simple">
<li><p>L_disjuncta_OR_Mt_Hood</p></li>
</ul>
</td>
<td><p>?</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_disjuncta_CA_Lily_Lake</p></li>
</ul>
</td>
<td><p>?</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>coalescentpop006</p></li>
</ul>
</td>
<td><p>?</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>coalescentpop008</p></li>
</ul>
</td>
<td><p>?</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>coalescentpop009</p></li>
</ul>
</td>
<td><p>?</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_lindrothi_CA_Emerald_Lake</p></li>
</ul>
</td>
<td><p>?</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>coalescentpop011</p></li>
</ul>
</td>
<td><p>?</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_lindrothi_CA_South_Fork_Bishop_Creek</p></li>
</ul>
</td>
<td><p>?</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_probata_UT_Shingle_Creek</p></li>
</ul>
</td>
<td><p>?</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>coalescentpop024</p></li>
</ul>
</td>
<td><p>?</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_tuulukwa_CA_Trinity_Alps</p></li>
</ul>
</td>
<td><p>?</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_tuulukwa_OR_Knowles_Creek</p></li>
</ul>
</td>
<td><p>?</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_tuulukwa_OR_Marys_Peak</p></li>
</ul>
</td>
<td><p>?</p></td>
<td><p>0</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="stage-iv-delimitation-of-species-units">
<h2>Stage IV. Delimitation of Species Units<a class="headerlink" href="#stage-iv-delimitation-of-species-units" title="Permalink to this headline">¶</a></h2>
<p>We are now ready to run DELINEATE to carry out the species delimitation analysis.
This will target the population lineages of unknown (“0”) status, and calculate the probabilities of all possible partitions that vary in their placement with respect to species identities.
The full analytical setup can be found in the subdirectory <span class="filepath">04-species-delimitation/</span>.
For input data, we will use the phylogeny obtained in Stage II (<span class="filepath">lionepha/03-population-tree-estimation/lionepha-p095-hkyg.mcct-mean-age.tree.nex</span>) and the constraints table in Stage III (<span class="filepath">04-species-delimitation/lionepha.run1.tsv</span>) as inputs to the program <span class="program">delineate-estimate</span>.
To execute the analysis, we run the following command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">delineate</span><span class="o">-</span><span class="n">estimate</span> <span class="n">partitions</span> \
    <span class="o">-</span><span class="n">P</span> <span class="mf">0.99</span> \
    <span class="o">-</span><span class="n">t</span> <span class="n">lionepha</span><span class="o">-</span><span class="n">p095</span><span class="o">-</span><span class="n">hkyg</span><span class="o">.</span><span class="n">mcct</span><span class="o">-</span><span class="n">mean</span><span class="o">-</span><span class="n">age</span><span class="o">.</span><span class="n">tree</span><span class="o">.</span><span class="n">nex</span> \
    <span class="o">-</span><span class="n">c</span> <span class="n">lionepha</span><span class="o">.</span><span class="n">run1</span><span class="o">.</span><span class="n">tsv</span>
</pre></div>
</div>
<p>The “-P 0.99” flag says to only report the probabilities of all partitions that contribute collectively to over 0.99 of the probabilities.
There could be thousands (or billions, depending on the number of unconstrained lineages you set) of other partitions in the 0.01 remaining tail that we choose to ignore to save disk space and time.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p><strong>Analysis Size and Computational Requirements</strong></p>
<p>This analysis has 13 unconstrained lineages out of 56.
The former (i.e., the number of unconstrained lineages) is the primary factor driving computational requirements, both in terms of time as well as memory: an analysis of 10,000 populations of which only 5 unconstrained will run faster (almost instantaneously) than one with 100 populations with 10 unconstrained.
An analysis with 50 unconstrained lineages will probably never complete even with the most powerful computers thrown at it, regardless of how few or many the contrained lineages are.
With this example of 13 unconstrained population lineages, we require 115 GB of computer memory and took about 35 mins to run on a fairly current (as of 2019) high-performance computing cluster.
If you find that this might be too demanding for your hardware (especially the memory requirement), you can reduce the complexity of the analysis by “fixing” the status of a few more species to be constraints by changing some “0’s” to “1’s” in the status columns of the constraint file (<span class="filepath">04-species-delimitation/lionepha.run1.tsv</span>).</p>
</div>
<p>The analysis will produce two files:</p>
<ul class="simple">
<li><p><span class="filepath">lionepha.run1.delimitation-results.json</span></p></li>
<li><p><span class="filepath">lionepha.run1.delimitation-results.trees</span></p></li>
</ul>
<p>There are included in the example archive in compressed form: <span class="filepath">04-species-delimitation/lionepha-delimitation-results.zip</span>.</p>
<p>The JSON-format file (<span class="filepath">lionepha.run1.delimitation-results.json</span>) lists some metadata about the analysis, followed by detailed description of all partitions considered.
The partitions are listed in order of descending probability, i.e., with the partition of highest probability given first.
Thus, the <em>maximum likelihood estimate</em> of the species delimitation is represented by the first partition listed.
The second partition listed is the species delimitation of the next highest probability, and so on.</p>
<p>The tree file consists of a set of trees, with one tree for each partition considered.
The tree structure itself — the phylogeny, including both topology and the branch lengths — are identical across all trees.
The only difference in the trees is the metadata markup of each tree, with annotations regarding species assignments of the tip (population) lineages.
Each tree corresponds to the a partition listed in the JSON file, in the same order.
Thus the trees illustrate the different partitions (and species assignments/boundaries of each partition), with the first tree representing the maximum likelihood estimate of the species delimitation, the second tree listed being the next most probable species delimitation and so on.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The results files are very large which may make viewing them challenging.
To facilitate viewing/manipulating of the most probable results, we provide a smaller tree file with just the first 10 trees here: <span class="filepath">04-species-delimitation/lionepha.run1.delimitation-results.trunc.trees</span>.
As the the result entries (trees and partitions) are given in order of descending probability, the earlier results have a higher probability than the later ones, so truncating the tree file by selecting the first few like this will let you view the most probable results.
In fact, the <em>first</em> listing in the results will have the highest probablity, and thus is MLE partition/tree.</p>
</div>
</div>
<div class="section" id="examination-and-interpretation-of-the-results">
<h2>Examination and Interpretation of the Results<a class="headerlink" href="#examination-and-interpretation-of-the-results" title="Permalink to this headline">¶</a></h2>
<p>The trees generated by DELINEATE have some special mark-up such that when opened in <a class="reference external" href="http://tree.bio.ed.ac.uk/software/figtree/"><span class="program">FigTree</span></a> will show some extra information.
For example, opening up the truncated trees result file (<span class="filepath">lionepha.run1.delimitation-results.trunc.trees</span>) in <a class="reference external" href="http://tree.bio.ed.ac.uk/software/figtree/"><span class="program">FigTree</span></a> and viewing the first tree, i.e. the maximum likelihood species delimitation shows the following:</p>
<div class="figure align-default">
<a class="figure-image reference internal image-reference" href="_images/lionepha.run1.delimitation-results.trunc.trees.jpg"><img alt="Results of species delimitation analysis." class="figure-image" src="_images/lionepha.run1.delimitation-results.trunc.trees.jpg" style="width: 100%;" /></a>
</div>
<p>The branches painted in <span class="constrained-branch-color">blue</span> (<span class="constrained-branch-color">▆</span>) represent the constrained population lineages, i.e., lineages with a fixed species assignment.
The branches painted in <span class="unconstrained-branch-color">ochre</span> (<span class="unconstrained-branch-color">▆</span>), on the other hand, represent the unconstrained population lineages, i.e., lineages with known identity and whose species membership was allowed to vary.
Furthermore the lineage labels are in a special format, showing the species assignment followed by the actual population label in parenthesis.</p>
<p>We interpret the results as follows:</p>
<table class="colwidths-given small-text compressed-table docutils align-default" id="id2">
<caption><span class="caption-text">Maximum Likelihood Species Delimitation</span><a class="headerlink" href="#id2" title="Permalink to this table">¶</a></caption>
<colgroup>
<col style="width: 15%" />
<col style="width: 15%" />
<col style="width: 70%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Lineages</p></th>
<th class="head"><p>Species Delimitation</p></th>
<th class="head"><p>Explanation</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><ul class="simple">
<li><p>coalescentpop008</p></li>
<li><p>coalescentpop009</p></li>
</ul>
</td>
<td><p><strong>DelineatedSp002</strong></p></td>
<td><p>Group together in the same <em>NEW</em> and unnamed species, labeled “<strong>DelineatedSp002</strong>”.
That is, the boundaries of the between these two lineages on the one hand, and all other lineages on the other, are <em>species</em> boundaries, not population.</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_lindrothi_CA_Emerald_Lake</p></li>
<li><p>coalescentpop011</p></li>
<li><p>L_lindrothi_CA_South_Fork_Bishop_Creek</p></li>
</ul>
</td>
<td><p><em>lindrothi</em></p></td>
<td><p>Group together as populations that are <em>part</em> of a pre-defined species, “<em>L. lindorothi</em>”.
That is, the boundaries of the between these three lineages on the one hand, and all other <em>lindrothi</em> lineages on the other, are (just) population boundaries, not species.</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_probata_UT_Shingle_Creek</p></li>
<li><p>coalescentpop024</p></li>
</ul>
</td>
<td><p><em>probata</em></p></td>
<td><p>Group together as populations that are <em>part</em> of a pre-defined species, “<em>L. probata</em>”.
That is, the boundaries of the between these two lineages on the one hand, and all other <em>probata</em> lineages on the other, are (just) population boundaries, not species.</p></td>
</tr>
<tr class="row-odd"><td><ul class="simple">
<li><p>L_disjuncta_OR_Mt_Hood</p></li>
<li><p>L_disjuncta_CA_Lily_Lake</p></li>
<li><p>coalescentpop006</p></li>
</ul>
</td>
<td><p><em>disjuncta</em></p></td>
<td><p>Group together as populations that are <em>part</em> of a pre-defined species, “<em>L. disjuncta</em>”.
That is, the boundaries of the between these three lineages on the one hand, and all other <em>disjuncta</em> lineages on the other, are (just) population boundaries, not species.</p></td>
</tr>
<tr class="row-even"><td><ul class="simple">
<li><p>L_tuulukwa_CA_Trinity_Alps</p></li>
<li><p>L_tuulukwa_OR_Knowles_Creek</p></li>
<li><p>L_tuulukwa_OR_Marys_Peak</p></li>
</ul>
</td>
<td><p><strong>DelineatedSp001</strong></p></td>
<td><p>Group together in the same <em>NEW</em> and unnamed species, labeled “<strong>DelineatedSp001</strong>”.
That is, the boundaries of the between these three lineages on the one hand, and all other lineages on the other, are <em>species</em> boundaries, not population.</p></td>
</tr>
</tbody>
</table>
<p>Conferring with the reference alpha taxonomy as worked out by Maddison and Poul (2020), we see that these are perfectly concordant.
The “DelineatedSp001” populations all collectively corresponding to the distinct species <em>L. tuulukwa</em> and “DelineatedSp002” populations all collectively corresponding to the distinct species <em>L. kavanaughi</em>, while the remaining unconstrained population lineages indeed were referred to <em>probata</em>, <em>lindrothi</em>, and <em>disjuncta</em> species.</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="quickstart.html" title="Quick Start"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">DELINEATE 1.2.3 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2020, Jeet Sukumaran and Mark T. Holder.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.4.0.
    </div>
  </body>
</html>